### DISCUSSION TITLE
Clip jpg image into transparent area

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#2731] [Help]
How can I clip a jpg image into a transparent area of a png like this:

![Screenshot_7](https://user-images.githubusercontent.com/48103248/96237550-ba9c0000-0f9d-11eb-9b6e-65a5b3c67b7d.jpg)

(IM 7 on Windows 10 cmd)

**Comment by snibgo** [comment]
You haven't shown your PNG with transparency. Instead, you show a JPG of a screenshot of that file. Being a JPG, it has no transparency. So I can't guide you through the process.

But the basic method is: compose your PNG with transparency over any image you want. You might need to resize that image, or crop it. And you might need a "-geometry" option.

**Comment by tbmuc** [comment]
Thank you very much. May I ask 2 more questions.

1) I uploaded a small JPG Mockup, because the images are usually 7000x5000 (40-50MB) . Is it ok, to upload large images here?
2) how to combine two compose lines into one:
magick composite -geometry +3900+870 %1-c32.jpg  -size 7086x5315 xc:#ffffff %1-tmp2.jpg
magick composite  %TEMPLATE% %1-tmp2.jpg  %1-tmp3.jpg

**Comment by snibgo** [comment]
1. Small files are better. We don't all have good internet service. For questions that are not actually about large images, I suggest a 1 MB limit.

2. I suggest using plain "magick" with a "-composite" operation, not "magick composite". So those commands become:
```
magick -size 7086x5315 xc:#ffffff %1-c32.jpg -geometry +3900+870 -composite %1-tmp2.jpg
magick %1-tmp2.jpg %TEMPLATE% -composite %1-tmp3.jpg
```
We can merge these into a single command:
```
magick -size 7086x5315 xc:#ffffff %1-c32.jpg -geometry +3900+870 -composite %TEMPLATE% +geometry -composite %1-tmp3.jpg
```
Note we need "+geometry" to cancel the previous "-geometry" setting.
